wakatime-review.js

```diff
@@ -1,7 +1,62 @@
+
+const bitbucket = {
+    readBranch: () => document.querySelector("#id_source_group .branch a").textContent,
+    readDomainOwnerAndProject: () => {
+        const url = window.location.href;
+        const regexp = /http[s]?:\/\/([a-zA-Z0-9]*\.[a-z]*)\/(\w*)\/([\w-]*).*/g;
+        const matched = regexp.exec(url);
+        const domain = matched[1];
+        const owner = matched[2];
+        const project = matched[3];
+        return {
+            domain: domain,
+            owner: owner,
+            project: project
+        };
+    }
+};
+
+
+const stash = {
+    readBranch: () => document.querySelector('div.pull-request-branches').textContent,
+    readDomainOwnerAndProject: () => {
+
+        let grabAfter = (array, after) => {
+            return array[array.lastIndexOf(after) + 1];
+        };
+
+        const url = new URL(window.location.href);
+        // example of URL to parse
+        // https://stash.clearcode.cc/projects/CCADS/repos/backend/pull-requests/137/commits
```

Comment:
ok. Now I see that for hosted solutions it might work, but for custom ones like yours we should have configuration. And I should definitely add support for github so I will now how much time I have spent on this review

Reply:
:+1: exactly. So this would require a bit of work on the configuration part.